packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -509,14 +568,28 @@ const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = isolatedFie
 		const fieldChange = change.fieldChange;
 		if (fieldChange !== undefined) {
 			inverse.fieldChange = { wasEmpty: fieldChange.newContent === undefined };
+			if (fieldChange.newContent?.changes !== undefined) {
+				// The node inserted by change will be the node deleted by inverse
+				// Move the inverted changes to the child change field
+				inverse.childChange = invertChild(fieldChange.newContent.changes, 0);
+			}
+
 			if (!fieldChange.wasEmpty) {
 				assert(revision !== undefined, 0x592 /* Unable to revert to undefined revision */);
-				inverse.fieldChange.newContent = { revert: reviver(revision, 0, 1)[0] };
+				inverse.fieldChange.newContent = { revert: reviver(revision, 0, 1)[0], revision };
+				if (change.childChange !== undefined) {
+					assert(change.deletedBy === undefined, "Unhandled scenario");
+					inverse.fieldChange.newContent.changes = invertChild(change.childChange, 0);
+				}
+			}
+		} else {
+			if (change.childChange !== undefined) {
+				inverse.childChange = invertChild(change.childChange, 0);
 			}
-		}
 
-		if (change.childChange !== undefined) {
-			inverse.childChange = invertChild(change.childChange, 0);
+			if (change.deletedBy !== undefined) {
+				inverse.deletedBy = change.deletedBy;
+			}
```

Comment:
When `change.childChange !== undefined` and `change.deletedBy !== undefined`, this will lead to the problem I described above (potentially undoing a change that's hasn't been applied).
I think it's both an easy trap to fall into and would be a pain to debug, so it would be good to add a test scenario for it.

You want the same kind of policy as above: if `change.deletedBy` is populated then just drop the changes.

Reply:
Latest commit should be dropping the changes if `change.deletedBy` is populated.